pkg/deployer/helm/ensure.go

```diff
@@ -74,40 +86,63 @@ func (h *Helm) ApplyFiles(ctx context.Context, files map[string]string, exports
 			},
 		}
 	}
+	h.ProviderStatus.ManagedResources = managedResources
 
-	if h.ProviderStatus != nil {
-		if err := h.cleanupOrphanedResources(ctx, targetClient, h.ProviderStatus.ManagedResources, objects); err != nil {
-			err = fmt.Errorf("unable to cleanup orphaned resources: %w", err)
-			return lsv1alpha1helper.NewWrappedError(err,
-				currOp, "CleanupOrphanedResources", err.Error())
-		}
+	if err := h.cleanupOrphanedResources(ctx, targetClient, h.ProviderStatus.ManagedResources, objects); err != nil {
+		err = fmt.Errorf("unable to cleanup orphaned resources: %w", err)
+		return lsv1alpha1helper.NewWrappedError(err,
+			currOp, "CleanupOrphanedResources", err.Error())
 	}
 
-	status := &helmv1alpha1.ProviderStatus{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: helmv1alpha1.SchemeGroupVersion.String(),
-			Kind:       "ProviderStatus",
-		},
-		ManagedResources: managedResources,
-	}
-	statusData, err := encodeStatus(status)
+	h.DeployItem.Status.ProviderStatus, err = encodeStatus(h.ProviderStatus)
 	if err != nil {
 		return lsv1alpha1helper.NewWrappedError(err,
-			currOp, "UpdateProviderStatus", err.Error())
+			currOp, "ProviderStatus", err.Error())
+	}
+	if err := h.kubeClient.Status().Update(ctx, h.DeployItem); err != nil {
+		return lsv1alpha1helper.NewWrappedError(err,
+			currOp, "UpdateStatus", err.Error())
 	}
 
 	if err := h.createOrUpdateExport(ctx, exports); err != nil {
 		return lsv1alpha1helper.NewWrappedError(err,
 			currOp, "CreateExport", err.Error())
 	}
 
-	h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
-	h.DeployItem.Status.ProviderStatus = statusData
-	h.DeployItem.Status.ObservedGeneration = h.DeployItem.Generation
-	if err := h.kubeClient.Status().Update(ctx, h.DeployItem); err != nil {
+	if h.ProviderConfiguration.HealthChecks.DisableDefault {
+		h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
+		h.DeployItem.Status.ObservedGeneration = h.DeployItem.Generation
+		h.DeployItem.Status.LastError = nil
+		return nil
+	}
+
+	return h.CheckResourcesHealth(ctx, targetClient)
```

Comment:
code here is duplicated.
Why not just check for an error in the checkresourceshealth func.
And remove lines 144-146.

 Like:
```suggestion
	if !h.ProviderConfiguration.HealthChecks.DisableDefault {
	  if err := h.CheckResourcesHealth(ctx, targetClient); err != nil {
	    // handle error
	  }
	}
	h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
        h.DeployItem.Status.ObservedGeneration = h.DeployItem.Generation
        h.DeployItem.Status.LastError = nil
        return nil
```

Reply:
Moved the code into the CheckResourcesHealth function and delegated most of that code into a defaultCheckResourcesHealth function to already provide the prerequisites to implement custom health checks.